### DISCUSSION TITLE
Vercel Date Issue

### DISCUSSION THREAD

**Discussion by sukrit101** [vercel/community#6477] [Help]
### Summary

I'm using Next.js here.
I understand that Vercel servers have different dates compared to the client's date. Typically, this is resolved by marking a component as 'use client' to generate dates on the client side.

However, I'm encountering an issue with my app's calendar feature. When the client sends a date to the server to fetch events, there's a discrepancy between the client's date and the server's date. This leads to events not being displayed correctly.

For example, if I request events for April 11th, but the event's date in the database falls between 5:29 AM and 11:59 PM, it won't be displayed for that day. However, if the client's date is April 10th, then the event shows up. Events with dates between 12:00 AM and 5:29 AM work fine.

Essentially, there's a 5 and a half hour gap causing this issue.

You can see the problem by visiting my [app:](https://scribe-rho.vercel.app/calendar).

To begin testing, please sign up if you haven't already, then navigate to the `/calendar` page. Create two events: one scheduled between 12:00 AM and 5:29 AM, and another scheduled between 5:30 AM and 11:59 PM.

I've also attached some images for reference.

![Screenshot 2024-04-11 224734](https://github.com/vercel/community/assets/150142378/446b04c8-e76c-416f-b3c5-857e541e24d1)
![image](https://github.com/vercel/community/assets/150142378/f6049153-f748-402b-b77c-016b837e35a8)
![image](https://github.com/vercel/community/assets/150142378/c287739e-ad0c-4fed-80e0-0381950d904d)

### Example

https://scribe-rho.vercel.app/calendar

### Steps to Reproduce

- Please sign up if you haven't already, then navigate to the `/calendar` page.
- Create two events: one scheduled between 12:00 AM and 5:29 AM, and another scheduled between 5:30 AM and 11:59 PM.

**Comment by versecafe** [comment]
@sukrit101 What time zone are you expecting? By default things should be using GMT which can create this gap or a very similar variant for users on the North American East coast.
